### DISCUSSION TITLE
Update effect on installed Ruby versions.

### DISCUSSION THREAD

**Discussion by ghost** [rbenv/rbenv#1485] [Q&A] [answered]
Hi,

I'm currently running an older version of rbenv on my production server and I would like to upgrade it to use ruby 3. 
I'm wondering if I upgrade rbenv, will it override my current versions of ruby installed. Or will my old versions still be installed. 
I still want to use the old versions and don't want my rails on my server to crash. 

Any help would be appreciated!

**Comment by jasonkarns** [comment, marked as answer]
The short answer is that upgrading rbenv itself should have no impact on the versions of ruby that are installed. Rubies live under `$(rbenv root)/versions` (see `RBENV_ROOT` under [Environment Variables](https://github.com/rbenv/rbenv#environment-variables) in readme)

Rbenv itself can live anywhere, with its `bin/` directory added to PATH. If you installed rbenv via git clone, then simply pulling would only update rbenv's installation directory. If you installed via a package manager, like homebrew, then it can be upgraded with the package manager and again would only affect rbenv's installation itself.

**Reply by ghost** [reply]
Thank you so much for your reply！
This helps me a lot!
